src/Draco.Compiler/Syntax/Parser.cs

```diff
@@ -32,5 +34,104 @@ public CompilationUnit ParseCompilationUnit()
         return new(decls.ToValue());
     }
 
-    private Decl ParseDeclaration() => throw new NotImplementedException();
+    private Decl ParseDeclaration()
+    {
+        var keyword = this.tokenSource.Peek();
+        if (keyword.Type == TokenType.KeywordFunc)
+        {
+            return this.ParseFunction();
+        }
+        else if (keyword.Type == TokenType.KeywordVal)
+        {
+            keyword = this.Expect(TokenType.KeywordVal);
+        }
+        else if (keyword.Type == TokenType.KeywordVar)
+        {
+            keyword = this.Expect(TokenType.KeywordVar);
+        }
+        var identifier = this.Expect(TokenType.Identifier);
+        TypeSpecifier? type = null;
+        if (this.tokenSource.Peek().Type == TokenType.Colon)
+        {
+            var colon = this.Expect(TokenType.Colon);
+            var typeIdentifier = this.Expect(TokenType.Identifier);
+            type = new TypeSpecifier(colon, new TypeExpr.Name(typeIdentifier));
+        }
+        if (this.tokenSource.Peek().Type == TokenType.Equal)
+        {
+            var equal = this.Expect(TokenType.Equal);
+            var value = this.ParseExpr();
+            this.Expect(TokenType.Semicolon);
+            return new Decl.Variable(keyword, identifier, type, (equal, value));
+        }
+        else
+        {
+            this.Expect(TokenType.Semicolon);
+            return new Decl.Variable(keyword, identifier, type, null);
+        }
+    }
+
+    private Func ParseFunction()
```

Comment:
Maybe `ParseFuncDeclaration` instead, later on we will likely have function expressions as well to deal with